components/server/src/server.ts

```diff
@@ -351,13 +356,29 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
         if (!this.config.runDbDeleter) {
             return;
         }
-        this.periodicDbDeleter.start(async () => {
-            const areWeLeader = await this.qorum.areWeLeader();
-            log.info(
-                "[PeriodicDbDeleter]" + areWeLeader ? "Deleter should run." : "Current instance is not the leader",
-            );
-            return areWeLeader;
-        });
+
+        const intervalMs = 30 * 1000;
+        repeat(async () => {
```

Comment:
Can we push this into the DB deleter? Feels like spoiling too much internals here.

Reply:
Not really. The redis implementation is on the `server` layer. Because we need the following wrapping: `repeat(mutex(job()))`, we'd either need to bring the mutex dependency into gitpod-db, or we need to keep the repeat in server.

Given this, can you think of a way to achieve what you want without pulling redis into the db package?

Reply:
> we'd either need to bring the mutex dependency into gitpod-db

Ah, overlooked that barrier. :bulb: 

> Given this, can you think of a way to achieve what you want without pulling redis into the db package?

We could always introduce indirection, the solution to all problems in SE :wink: 
A simple method in `server` would already be great.

Reply:
In what way is that different to the method `startDbDeleter` here? I'm not sure I fully understand what it should look like.

Yep, indirection could work but in general I felt this is cleaner/easier to read (given our singular call-site) than introducing the indirection. Not to mention, we plan to revisit the periodic deleter anyway

Reply:
Oh, forgot what I said. Should expand to more context... :see_no_evil: :stuck_out_tongue: 

SORRY!